web/src/filter.ts

```diff
@@ -791,9 +792,60 @@ export class Filter {
         return [...parts, ...more_parts];
     }
 
+    // Generates a descriptive phrase based on the type and operand of a given part
+    static generateDescription(part: Part): string | undefined {
+        switch (part.type) {
+            case "plain_text":
+                return part.content;
+            case "channel_topic":
+                return $t({defaultMessage: "channel {channel} > {topic}"}, part);
+            case "invalid_has":
+                return $t({defaultMessage: "invalid {operand} operand for has operator"}, part);
+            case "prefix_for_operator":
+                part.operand +=
+                    part.operand === "link" ||
+                    part.operand === "image" ||
+                    part.operand === "attachment" ||
+                    part.operand === "reaction"
+                        ? "s"
+                        : "";
+                return $t({defaultMessage: "{prefix_for_operator} {operand}"}, part);
+            case "is_operator":
+                switch (part.operand) {
```

Comment:
This still feels a bit odd to have operator/operand substitutions here. I wonder if its better to have full string(?). Or maybe you have a better idea with this approach